graph/push_v2.go

```diff
@@ -114,13 +114,17 @@ func (p *v2Pusher) pushV2Tag(tag string) error {
 	}
 
 	out := p.config.OutStream
+	layersSeen := make(map[string]bool)
 
 	for ; layer != nil; layer, err = p.graph.GetParent(layer) {
 		if err != nil {
 			return err
 		}
 
-		if p.layersSeen[layer.ID] {
+		// break early if layer has already been seen in this image,
+		// this prevents infinite loops on layers which loopback, this
+		// cannot be prevented since layer IDs are not merkle hashes
+		if layersSeen[layer.ID] {
```

Comment:
I was actually wondering the same thing. We would have to be confident that there are no valid cases which would trigger this condition. While I think it is fine, would require some due diligence, and out of scope for this commit. I can add a TODO in the comment though